File: 53-maximum-subarray/53-maximum-subarray.py

class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        size = len(nums)
        max_so_far =nums[0]
        curr_max = nums[0]
     
        for i in range(1,size):
            curr_max = max(nums[i], curr_max + nums[i])
            max_so_far = max(max_so_far,curr_max)
         
        return max_so_far
             
        

File: 53-maximum-subarray/53-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int size = nums.size();
        int max_so_far = nums[0];
        int curr_so_far = nums[0];
        for(int i = 1;i<size;i++){
            curr_so_far = max(nums[i], curr_so_far+nums[i]);
            max_so_far = max(max_so_far,curr_so_far);
        }
        return max_so_far;
    }
};